src/simple_commander/main.py

```diff
@@ -220,21 +229,24 @@ def hit(self, other_unit):
         unit_class_name = other_unit. __class__.__name__
         logging.info('In hit - %s and %s' % (self.__class__.__name__, unit_class_name))
         if unit_class_name == 'Hero':
+            other_unit.hits += 1
             other_unit.decrease_life()
         else:
+            if other_unit.__class__.__name__ == 'Bullet':
```

Comment:
if isinstance(other_unit, Bullet):
--or--
if other_unit is Bullet

Anyway, this logic must be part of GameController in future